interfaces/repo.go

```diff
@@ -726,6 +726,17 @@ func (r *Repository) DisconnectSnap(snapName string) ([]string, error) {
 	return result, nil
 }
 
+// isSpecialAutoConnectSnap checks special Name/Developer combinations to see
+// if this particular snap's connections should be automatically connected even
+// if the interfaces are not autoconnect and the snap is not an OS snap.
+// TODO: When assertions are in place this code should be removed.
+func isSpecialAutoConnectSnap(snap *snap.Info) bool {
+	if snap.Name() == "canonical-livepatch" && snap.Developer == "canonical" {
```

Comment:
@chipaca I believe the check for "canonical" already sorts that out. We only get the developer filed when it comes from the store, right?